sql/hive/pom.xml

```diff
@@ -177,6 +177,10 @@
       <artifactId>libfb303</artifactId>
     </dependency>
     <dependency>
+      <groupId>org.apache.derby</groupId>
+      <artifactId>derby</artifactId>
```

Comment:
Hive metastore depends on derby 10.10.2, and we package derby 10.12.1 when building Spark, so at last we are using derby 10.12.1 when Spark uses local hive metastore.

However this is a tricky approach, e.g. it doesn't work for SBT. When you build Spark with SBT, you are still using derby 10.10.2. It's probably the reason why the test failed on jenkins.

Here I explicitly add the derby dependency to the hive module, to overwrite the default derby 10.10.2 dependency.

Reply:
I see. Thank you!